core/src/test/java/com/rudderstack/core/CentralPluginChainTest.kt

```diff
@@ -16,132 +16,109 @@ package com.rudderstack.core
-        assertThat(msg, not(it.originalMessage))
-        //destination plugins should always call proceed on original message to discard changes.
-        it.proceed(it.originalMessage)
-    }
-    private val dummyPlugin3 = Plugin {
-        val msg = it.message()
-        assertThat(enteredPluginNo ++, `is`(MAIN_PLUGIN_3_SL_NO))
-        it.proceed(msg)
-    }*/
     @Before
-    fun setup(){
+    fun setup() {
+        MockKAnnotations.init(this, relaxed = true)
         // Mock the behavior of plugins and destination plugin
-        whenever(mockPlugin1.intercept(any())).thenAnswer{
-            val chain = it.arguments[0] as CentralPluginChain
+        every { mockPlugin1.intercept(any()) } answers {
+            val chain = arg<CentralPluginChain>(0)
             chain.proceed(mockMessage)
         }
-        whenever(mockPlugin2.intercept(any())).thenAnswer{
-            val chain = it.arguments[0] as CentralPluginChain
+        every { mockPlugin2.intercept(any()) } answers {
+            val chain = arg<CentralPluginChain>(0)
             chain.proceed(mockMessage)
         }
-        whenever(mockDestinationPlugin.intercept(any())).thenAnswer{
-            val chain = it.arguments[0] as CentralPluginChain
+        every { mockDestinationPlugin.intercept(any()) } answers {
+            val chain = arg<CentralPluginChain>(0)
             chain.proceed(mockMessage)
         }
+
         // Initialize the list of plugins for testing
         val plugins = listOf(mockPlugin1, mockPlugin2, mockDestinationPlugin)
         centralPluginChain = CentralPluginChain(mockMessage, plugins, originalMessage = mockMessage)
-        whenever(mockMessage.copy()).thenReturn(mockMessage)
+        every { mockMessage.copy() } returns mockMessage
     }
+
     @Test
     fun testMessage() {
         assertThat(centralPluginChain.message(), equalTo(mockMessage))
     }
 
     @Test
     fun testProceed() {
+        // Verify interactions
+        val chainCaptor1 = slot<CentralPluginChain>()
+        every { mockPlugin1.intercept(capture(chainCaptor1)) } answers {
+            val chain = arg<CentralPluginChain>(0)
+            chain.proceed(mockMessage)
+        }
 
+        val chainCaptor2 = slot<CentralPluginChain>()
+        every { mockPlugin2.intercept(capture(chainCaptor2)) } answers {
+            val chain = arg<CentralPluginChain>(0)
+            chain.proceed(mockMessage)
+        }
+
+
+        val chainCaptor3 = slot<CentralPluginChain>()
+        every { mockDestinationPlugin.intercept(capture(chainCaptor3)) } answers {
+            val chain = arg<CentralPluginChain>(0)
+            chain.proceed(mockMessage)
+        }
 
         // Call the method under test
         val resultMessage = centralPluginChain.proceed(mockMessage)
 
-        // Verify interactions
-        val chainCaptor1 = argumentCaptor<CentralPluginChain>()
-        verify(mockPlugin1).intercept(chainCaptor1.capture())
```

Comment:
We need the verify tests

Reply:
In general we don't need to verify the action that happens to capture the value. The `intercept() `happens anyway and we capture the needed value and the test proceeds to the assertion.

Reply:
Maybe we need to have a more descriptive test name here, `testProceed()` does not really explain if we want to test something in particular.

For example:
```
    @Test
    fun  `given plugins exist in the chain, when message intercept happens, then the message is the expected`(){
    ...
    }
```
 or like 
```
    @Test
    fun  `when plugins exist in the chain, verify that message intercept happens`(){
    ...
    }
```

 In that case it will be clearer what we need to achieve. Let me know what you think.